ShopperInsights/src/main/java/com/braintreepayments/api/EligiblePaymentsApi.kt

```diff
@@ -3,6 +3,7 @@ package com.braintreepayments.api
 import com.braintreepayments.api.EligiblePaymentsApiRequest.Companion.toJson
 import org.json.JSONException
 
+@ExperimentalBetaApi
```

Comment:
Is there a benefit in marking these internal classes with the annotation?

Reply:
It is required by the compiler.

A class that depends on a @ExperimentalBetaApi annotated class can either be annotated with @ExperimentalBetaApi annotation or @OptIn(ExperimentalBetaApi::class).

Maybe in this case, the internal class can be annotated with @OptIn instead. Thoughts?

Reply:
Would the depednency go the other way for the internal classes? The `EligiblePaymentsApi` class would not depend on the `ShopperInsightsClient` class. I'm wondering if we can just mark the externally facing classes with the annotation. Also not sure if missing something 😄

Reply:
I've made the internal classes and tests OptIn to the functionality.
Even among the classes that are exposed, only those that I think can change are being marked with the ExperimentalBetaApi annotation. For eg.: The `ShopperInsightsResult` class now opts in to the experimental api and does _not_ propagate it to its usage location.

Reply:
got it. that sounds good!